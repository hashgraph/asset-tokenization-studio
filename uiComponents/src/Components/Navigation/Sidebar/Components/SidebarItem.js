var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import React from "react";
import { forwardRef, useMultiStyleConfig as useChakraMultiStyleConfig, } from "@chakra-ui/system";
import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";
import { Text } from "@/Components/Foundations/Text";
import { useHover } from "@/Hooks";
import { Center as ChakraCenter, Flex as ChakraFlex, useMergeRefs as useChakraMergeRefs, } from "@chakra-ui/react";
export var sidebarPartsList = ["iconWeight", "icon", "label", "container"];
export var SidebarItem = forwardRef(function (_a, ref) {
    var isActive = _a.isActive, isDisabled = _a.isDisabled, label = _a.label, icon = _a.icon, props = __rest(_a, ["isActive", "isDisabled", "label", "icon"]);
    var _b = useHover(), containerRef = _b[0], isHovered = _b[1];
    var refs = useChakraMergeRefs(containerRef, ref);
    var styles = useChakraMultiStyleConfig("SidebarItem", {
        isHovered: isHovered,
        isActive: isActive,
        isDisabled: isDisabled
    });
    return (React.createElement(ChakraFlex, __assign({ align: "center", direction: "column", outline: "none", "data-testid": "sidebar-item-".concat(label), disabled: isDisabled, ref: refs }, props, { sx: styles.container }),
        React.createElement(ChakraCenter, { as: "button", sx: styles.icon },
            React.createElement(PhosphorIcon, { size: "xs", as: icon, weight: styles.iconWeight })),
        React.createElement(Text, { sx: styles.label },
            label,
            " ")));
});
