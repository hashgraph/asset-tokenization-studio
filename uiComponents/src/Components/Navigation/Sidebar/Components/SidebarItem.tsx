import React from "react";
import {
  forwardRef,
  useMultiStyleConfig as useChakraMultiStyleConfig,
} from "@chakra-ui/system";
import type {
  PhosphorIconProps,
  Weight,
} from "@/Components/Foundations/PhosphorIcon";
import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";
import { Text } from "@/Components/Foundations/Text";
import { useHover } from "@/Hooks";
import type { FlexProps as ChakraFlexProps } from "@chakra-ui/react";
import {
  Center as ChakraCenter,
  Flex as ChakraFlex,
  useMergeRefs as useChakraMergeRefs,
} from "@chakra-ui/react";
import type {
  SystemStyleObject as ChakraSystemStyleObject,
  ComponentWithAs,
} from "@chakra-ui/system";
import type {
  BaseMultiStyleConfiguration,
  PartsStyleInterpolation,
} from "@/Theme/Components/BaseMultiStyleConfiguration";
export const sidebarPartsList: Array<
  "iconWeight" | "icon" | "label" | "container"
> = ["iconWeight", "icon", "label", "container"];

type Parts = typeof sidebarPartsList;

export type SidebarItemConfigProps = {
  isDisabled?: boolean;
  isHovered?: boolean;
  isActive?: boolean;
};

export interface SidebarItemThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {
  baseStyle:
    | (({
        isHovered,
        isDisabled,
        isActive,
      }: SidebarItemConfigProps) => Partial<SidebarItemThemeStyle>)
    | PartsStyleInterpolation<Parts>;
}

export interface SidebarItemProps extends ChakraFlexProps {
  isActive?: boolean;
  isDisabled?: boolean;
  label: string;
  icon: PhosphorIconProps["as"];
}

export type SidebarItemThemeStyle = Record<
  Parts[number],
  ChakraSystemStyleObject
> & {
  iconWeight: Weight;
};

export const SidebarItem: ComponentWithAs<"button", SidebarItemProps> =
  forwardRef<SidebarItemProps, "button">(
    (
      { isActive, isDisabled, label, icon, ...props }: SidebarItemProps,
      ref
    ) => {
      const [containerRef, isHovered] = useHover();
      const refs = useChakraMergeRefs(containerRef, ref);
      const styles = useChakraMultiStyleConfig("SidebarItem", {
        isHovered,
        isActive,
        isDisabled,
      }) as SidebarItemThemeStyle;

      return (
        <ChakraFlex
          align="center"
          direction="column"
          outline="none"
          // as="button"
          data-testid={`sidebar-item-${label}`}
          disabled={isDisabled}
          ref={refs}
          {...props}
          sx={styles.container}
        >
          <ChakraCenter as="button" sx={styles.icon}>
            <PhosphorIcon size="xs" as={icon} weight={styles.iconWeight} />
          </ChakraCenter>
          <Text sx={styles.label}>{label} </Text>
        </ChakraFlex>
      );
    }
  );
