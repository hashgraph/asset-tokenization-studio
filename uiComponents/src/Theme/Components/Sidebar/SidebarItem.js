var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { Weight } from "@/Components/Foundations/PhosphorIcon";
import { focusVisible } from "@/Theme/utils";
import { sidebarPartsList } from "@Components/Navigation/Sidebar";
var iconActiveStyle = {
    bg: "primary.100",
    color: "neutral.900"
};
var iconHoverStyle = {
    bg: "neutral.200"
};
var iconDisabledStyle = {
    color: "neutral.200"
};
var labelActiveStyle = {
    textStyle: "ElementsSemiboldXS",
    color: "neutral.900"
};
var labelDisabledStyle = {
    color: "neutral.200"
};
export var ConfigSidebarItem = {
    parts: sidebarPartsList,
    baseStyle: function (_a) {
        var isHovered = _a.isHovered, isActive = _a.isActive, isDisabled = _a.isDisabled;
        var isItemActive = isActive && !isDisabled;
        return {
            iconWeight: isActive && !isDisabled ? Weight.Fill : Weight.Regular,
            icon: __assign(__assign(__assign({ _focusVisible: focusVisible, borderColor: "red", color: "neutral.800", p: 1, borderRadius: "8px", transition: "all .3s ease", svg: {
                    color: "inherit"
                } }, (isHovered && iconHoverStyle)), (isItemActive && iconActiveStyle)), (isDisabled && iconDisabledStyle)),
            label: __assign(__assign({ transition: "all .3s ease", color: "neutral.800", textStyle: "ElementsMediumXS", mt: 1, px: 1 }, (isItemActive && labelActiveStyle)), (isDisabled && labelDisabledStyle)),
            container: {
                cursor: isDisabled ? "not-allowed" : "pointer",
                px: 2
            }
        };
    }
};
