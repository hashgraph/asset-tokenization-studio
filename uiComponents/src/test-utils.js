import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
