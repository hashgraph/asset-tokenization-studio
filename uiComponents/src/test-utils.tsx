import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import type { RenderResult } from "@testing-library/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";

const DefaultTheme = extendTheme(BasePlatformTheme);

export const render = (ui: JSX.Element, theme = DefaultTheme): RenderResult => {
  const wrapper = ({ children }: { children?: React.ReactNode }) => (
    <ChakraProvider theme={theme}>{children}</ChakraProvider>
  );

  return testingLibraryRender(ui, { wrapper });
};
