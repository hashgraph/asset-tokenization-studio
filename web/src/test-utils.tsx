import React from "react";
import { ChakraProvider } from "@chakra-ui/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import type { RenderOptions, RenderResult } from "@testing-library/react";
import { render, waitFor } from "@testing-library/react";
import type { MemoryHistory } from "history";
import { createMemoryHistory } from "history";
import { I18nextProvider } from "react-i18next";
import { Router } from "react-router-dom";
import i18n from "./i18n";
import theme from "./theme";
import userEvent from "@testing-library/user-event";

export const selectCalendar = async (
  component: RenderResult,
  id: string,
  day: string | number = new Date().getDate(),
) => {
  const calendar = component.getByTestId(id);
  await userEvent.click(calendar);

  await waitFor(() => {
    const daysToSelect = component.getAllByTestId(`day-${day}`);

    const dayToSelect = daysToSelect.find(
      (day) => !day.hasAttribute("disabled"),
    );

    if (dayToSelect) {
      userEvent.click(dayToSelect);
    }
  });

  await userEvent.click(document.body);
};

export const testQueryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      refetchOnMount: true,
      cacheTime: 0,
    },
  },
  logger: {
    log: console.log,
    warn: console.warn,
    // âœ… no more errors on the console for tests
    error: () => {},
  },
});

export const queryWrapper = ({ children }: { children: React.ReactNode }) => (
  <QueryClientProvider client={testQueryClient}>{children}</QueryClientProvider>
);

const memoryHistory = createMemoryHistory();

export const AllProviders = ({
  children,
  history = memoryHistory,
}: {
  children?: React.ReactNode;
  history?: MemoryHistory;
}) => {
  return (
    <I18nextProvider i18n={i18n}>
      <QueryClientProvider client={testQueryClient}>
        <ChakraProvider theme={theme}>
          <Router location={history.location} navigator={history}>
            {children}
          </Router>
        </ChakraProvider>
      </QueryClientProvider>
    </I18nextProvider>
  );
};

const customRender = (
  ui: React.ReactElement,
  {
    options,
    history,
  }: {
    options?: RenderOptions;
    history?: MemoryHistory;
  } = {},
): RenderResult =>
  render(ui, {
    wrapper: ({ children }) => (
      <AllProviders history={history}>{children}</AllProviders>
    ),
    ...options,
  });

export { customRender as render };
