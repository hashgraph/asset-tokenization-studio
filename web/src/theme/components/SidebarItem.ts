import {
  SidebarItemConfigProps,
  SidebarItemThemeConfiguration,
  sidebarPartsList,
} from "@hashgraph/uiComponents/Navigation";

const iconStyles = {
  hover: {
    bgColor: "neutral.150",
  },

  selected: {
    bgColor: "neutral.100",
    color: "neutral.650",
  },

  disabled: {
    color: "neutral.600",
  },
};

const labelStyles = {
  selected: {
    textStyle: "ElementsSemiboldXS",
  },

  disabled: {
    color: "neutral.600",
  },
};

export const SidebarItem: SidebarItemThemeConfiguration = {
  parts: sidebarPartsList,
  baseStyle: ({ isHovered, isActive, isDisabled }: SidebarItemConfigProps) => {
    const isSelected = isActive && !isDisabled;

    return {
      icon: {
        color: "neutral.800",
        ...(isSelected && iconStyles.selected),
        ...(isHovered && !isSelected && iconStyles.hover),
        ...(isDisabled && iconStyles.disabled),
      },

      label: {
        textStyle: "ElementsMediumXS",
        color: "neutral.800",
        ...(isSelected && labelStyles.selected),
        ...(isDisabled && labelStyles.disabled),
      },
    };
  },
};
