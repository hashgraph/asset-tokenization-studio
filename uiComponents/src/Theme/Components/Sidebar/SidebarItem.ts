import { Weight } from "@/Components/Foundations/PhosphorIcon";
import { focusVisible } from "@/Theme/utils";
import type { SidebarItemThemeConfiguration } from "@Components/Navigation/Sidebar";
import { sidebarPartsList } from "@Components/Navigation/Sidebar";

const iconActiveStyle = {
  bg: "primary.100",
  color: "neutral.900",
};
const iconHoverStyle = {
  bg: "neutral.200",
};
const iconDisabledStyle = {
  color: "neutral.200",
};

const labelActiveStyle = {
  textStyle: "ElementsSemiboldXS",
  color: "neutral.900",
};

const labelDisabledStyle = {
  color: "neutral.200",
};

export const ConfigSidebarItem: SidebarItemThemeConfiguration = {
  parts: sidebarPartsList,
  baseStyle: ({ isHovered, isActive, isDisabled }) => {
    const isItemActive = isActive && !isDisabled;

    return {
      iconWeight: isActive && !isDisabled ? Weight.Fill : Weight.Regular,
      icon: {
        _focusVisible: focusVisible,
        borderColor: "red",
        color: "neutral.800",
        p: 1,
        borderRadius: "8px",
        transition: "all .3s ease",
        svg: {
          color: "inherit",
        },
        ...(isHovered && iconHoverStyle),
        ...(isItemActive && iconActiveStyle),
        ...(isDisabled && iconDisabledStyle),
      },
      label: {
        transition: "all .3s ease",
        color: "neutral.800",
        textStyle: "ElementsMediumXS",
        mt: 1,
        px: 1,
        ...(isItemActive && labelActiveStyle),
        ...(isDisabled && labelDisabledStyle),
      },
      container: {
        cursor: isDisabled ? "not-allowed" : "pointer",
        px: 2,
      },
    };
  },
};
