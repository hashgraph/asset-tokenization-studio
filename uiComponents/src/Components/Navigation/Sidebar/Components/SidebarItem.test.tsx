import { render } from "@/test-utils";
import { Cookie } from "@phosphor-icons/react";
import React from "react";
import type { SidebarItemProps } from "./SidebarItem";
import { SidebarItem } from "./SidebarItem";
import { fireEvent, waitFor } from "@testing-library/react";

describe(`<SidebarItem />`, () => {
  const defaultProps: SidebarItemProps = {
    label: "Item",
    icon: Cookie,
  };
  const factoryComponent = (props: SidebarItemProps = defaultProps) =>
    render(<SidebarItem {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent();

    expect(component.asFragment()).toMatchSnapshot();
  });

  test("renders correctly when hovered", async () => {
    const component = factoryComponent();
    fireEvent.mouseEnter(
      component.getByTestId(`sidebar-item-${defaultProps.label}`)
    );

    await waitFor(() => {
      expect(component.asFragment()).toMatchSnapshot("Hovered");
    });
  });

  test("renders correctly when it's disabled", async () => {
    const component = factoryComponent({ ...defaultProps, isDisabled: true });

    await waitFor(() => {
      expect(component.asFragment()).toMatchSnapshot("Disabled");
    });
  });
  test("renders correctly when it's active", async () => {
    const component = factoryComponent({ ...defaultProps, isActive: true });

    await waitFor(() => {
      expect(component.asFragment()).toMatchSnapshot("Active");
    });
  });
});
