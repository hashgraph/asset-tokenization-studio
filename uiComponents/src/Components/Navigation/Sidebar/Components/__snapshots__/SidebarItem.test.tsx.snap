// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarItem /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-df6r3v"
    data-testid="sidebar-item-Item"
  >
    <button
      class="css-1rb62th"
    >
      <svg
        class="chakra-icon css-1qzovvq"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M164.49,163.51a12,12,0,1,1-17,0A12,12,0,0,1,164.49,163.51Zm-81-8a12,12,0,1,0,17,0A12,12,0,0,0,83.51,155.51Zm9-39a12,12,0,1,0-17,0A12,12,0,0,0,92.49,116.49Zm48-1a12,12,0,1,0,0,17A12,12,0,0,0,140.49,115.51ZM232,128A104,104,0,1,1,128,24a8,8,0,0,1,8,8,40,40,0,0,0,40,40,8,8,0,0,1,8,8,40,40,0,0,0,40,40A8,8,0,0,1,232,128Zm-16.31,7.39A56.13,56.13,0,0,1,168.5,87.5a56.13,56.13,0,0,1-47.89-47.19,88,88,0,1,0,95.08,95.08Z"
        />
      </svg>
    </button>
    <p
      class="css-owi0ee"
    >
      Item 
    </p>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<SidebarItem /> renders correctly when hovered: Hovered 1`] = `
<DocumentFragment>
  <div
    class="css-df6r3v"
    data-testid="sidebar-item-Item"
  >
    <button
      class="css-ek0iur"
    >
      <svg
        class="chakra-icon css-1qzovvq"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M164.49,163.51a12,12,0,1,1-17,0A12,12,0,0,1,164.49,163.51Zm-81-8a12,12,0,1,0,17,0A12,12,0,0,0,83.51,155.51Zm9-39a12,12,0,1,0-17,0A12,12,0,0,0,92.49,116.49Zm48-1a12,12,0,1,0,0,17A12,12,0,0,0,140.49,115.51ZM232,128A104,104,0,1,1,128,24a8,8,0,0,1,8,8,40,40,0,0,0,40,40,8,8,0,0,1,8,8,40,40,0,0,0,40,40A8,8,0,0,1,232,128Zm-16.31,7.39A56.13,56.13,0,0,1,168.5,87.5a56.13,56.13,0,0,1-47.89-47.19,88,88,0,1,0,95.08,95.08Z"
        />
      </svg>
    </button>
    <p
      class="css-owi0ee"
    >
      Item 
    </p>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<SidebarItem /> renders correctly when it's active: Active 1`] = `
<DocumentFragment>
  <div
    class="css-df6r3v"
    data-testid="sidebar-item-Item"
  >
    <button
      class="css-51z7wg"
    >
      <svg
        class="chakra-icon css-1qzovvq"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224,120a40,40,0,0,1-40-40,8,8,0,0,0-8-8,40,40,0,0,1-40-40,8,8,0,0,0-8-8A104,104,0,1,0,232,128,8,8,0,0,0,224,120ZM75.51,99.51a12,12,0,1,1,0,17A12,12,0,0,1,75.51,99.51Zm25,73a12,12,0,1,1,0-17A12,12,0,0,1,100.49,172.49Zm23-40a12,12,0,1,1,17,0A12,12,0,0,1,123.51,132.49Zm41,48a12,12,0,1,1,0-17A12,12,0,0,1,164.49,180.49Z"
        />
      </svg>
    </button>
    <p
      class="css-atzlkw"
    >
      Item 
    </p>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<SidebarItem /> renders correctly when it's disabled: Disabled 1`] = `
<DocumentFragment>
  <div
    class="css-1m0okgb"
    data-testid="sidebar-item-Item"
    disabled=""
  >
    <button
      class="css-u61f6f"
    >
      <svg
        class="chakra-icon css-1qzovvq"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M164.49,163.51a12,12,0,1,1-17,0A12,12,0,0,1,164.49,163.51Zm-81-8a12,12,0,1,0,17,0A12,12,0,0,0,83.51,155.51Zm9-39a12,12,0,1,0-17,0A12,12,0,0,0,92.49,116.49Zm48-1a12,12,0,1,0,0,17A12,12,0,0,0,140.49,115.51ZM232,128A104,104,0,1,1,128,24a8,8,0,0,1,8,8,40,40,0,0,0,40,40,8,8,0,0,1,8,8,40,40,0,0,0,40,40A8,8,0,0,1,232,128Zm-16.31,7.39A56.13,56.13,0,0,1,168.5,87.5a56.13,56.13,0,0,1-47.89-47.19,88,88,0,1,0,95.08,95.08Z"
        />
      </svg>
    </button>
    <p
      class="css-t8zpnn"
    >
      Item 
    </p>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
